src/main/java/htsjdk/tribble/util/ParsingUtils.java

```diff
@@ -95,22 +88,27 @@ public static InputStream openInputStream(String path)
      *
      * the wrapper will only be applied to paths that are not http, https, ftp, or file, i.e. any {@link java.nio.file.Path}
      * using a custom filesystem plugin
-     * @param path a uri like string
+     * @param uri a uri like string
      * @param wrapper to wrap the input stream in, may be used to implement caching or prefetching, etc
-     * @return
-     * @throws IOException
+     * @return An inputStream appropriately created from uri and conditionally wrapped with wrapper (only in certain cases
+     * @throws IOException when stream cannot be opened against uri
      */
-    public static InputStream openInputStream(String path, Function<SeekableByteChannel, SeekableByteChannel> wrapper)
+    public static InputStream openInputStream(final String uri, final Function<SeekableByteChannel, SeekableByteChannel> wrapper)
             throws IOException {
 
+        // will be empty string if there's no scheme
+        final String scheme = Optional.ofNullable(IOUtil.getScheme(uri)).orElse("");
+
         final InputStream inputStream;
-        if (path.startsWith("http:") || path.startsWith("https:") || path.startsWith("ftp:")) {
-            inputStream = getURLHelper(new URL(path)).openInputStream();
-        } else if (IOUtil.hasScheme(path)) {
-            inputStream = new SeekablePathStream(IOUtil.getPath(path), wrapper);
-        } else {
-            File file = new File(path);
+        final Set<String> URL_SCHEMES = new HashSet<>(Arrays.asList("http","ftp","https"));
+
+        if (URL_SCHEMES.contains(scheme)) {
+            inputStream = getURLHelper(new URL(uri)).openInputStream();
+        } else if (scheme.isEmpty()) {
+            File file = new File(uri);
             inputStream = new FileInputStream(file);
```

Comment:
when possible use `Files.newInputStream` instead of `new FileInputStream(` ( see https://www.cloudbees.com/blog/fileinputstream-fileoutputstream-considered-harmful )

Reply:
ok.